src/aria2p/api.py

```diff
@@ -850,3 +849,53 @@ def stop_listening(self) -> None:
         if self.listener:
             self.listener.join()
         self.listener = None
+
+    def split_input_file(self, lines):
+        """
+        Helper to split downloads in an input file.
+
+        Arguments:
+             lines:  The lines of the input file.
+        """
+        block = []
+        for line in lines:
+            if line.lstrip().startswith("#"):  # Ignore Comments
+                continue
+            if not line.strip():  # Ignore empty line
+                continue
+            if not line.startswith(" "):  # Read uri
+                if block:
+                    yield block
+                    block = []
+            block.append(line.rstrip("\n"))
+        if block:
+            yield block
+
+    def parse_input_file(self, input_file):
+        """
+        Parse a file with URIs or an aria2c input file.
+
+        Arguments:
+            input_file: Path to file with URIs or aria2c input file.
+
+        Returns:
+            List of tuples containing list of uris and dictionary with aria2c options.
+
+        """
+
+        downloads = []
+        with Path(input_file).open() as fd:
+            for download_lines in self.split_input_file(fd):
+                uris = download_lines[0].split("\t")
+                options = {}
+                try:
+                    for opt_line in download_lines[1:]:
+                        opt_name, opt_value = opt_line.split("=", 1)
+                        options[opt_name.lstrip()] = opt_value
+                    downloads.append((uris, options))
+                except ValueError as error:
+                    logger.error(f"Skipping download because of invalid option line '{opt_line}'")
+                    logger.opt(exception=True).trace(error)
+                    return []
```

Comment:
If we say "skipping download" (singular), we should at least return the other valid downloads.
If we indeed return nothing, we should say "Invalid input line (option line): {opt_line}" instead.
What do you think?

Reply:
Okay. Now I get it. The only download with an invalid option should be skipped, not the entire download queue. I'll correct this.